tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -134,20 +144,36 @@ def get_install_paths(duthost, fw, versions, chassis, target_component):
     for comp, revs in component.items():
         if comp in ver:
             if revs[0].get("upgrade_only", False) and ver[comp] not in [r["version"] for r in revs]:
-                log.warning("Firmware is upgrade only and existing firmware {} is not present in version list. Skipping {}".format(ver[comp], comp))
+                logger.warning("Firmware is upgrade only and existing firmware {} is not present in version list. "
+                               "Skipping {}".format(ver[comp], comp))
                 continue
             for i, rev in enumerate(revs):
                 if "hw_revision" in rev and rev["hw_revision"] != get_hw_revision(duthost):
-                    log.warning("Firmware {} only supports HW Revision {} and this chassis is {}. Skipping".format(rev["version"], rev["hw_revision"], get_hw_revision(duthost)))
+                    logger.warning("Firmware {} only supports HW Revision {} and this chassis is {}. Skipping".
+                                   format(rev["version"], rev["hw_revision"], get_hw_revision(duthost)))
                     continue
                 if rev["version"] != ver[comp]:
                     paths[comp] = rev
                     break
                 elif rev.get("upgrade_only", False):
-                    log.warning("Firmware is upgrade only and newer version than {} is not available. Skipping {}".format(ver[comp], comp))
+                    logger.warning("Firmware is upgrade only and newer version than {} is not available. Skipping {}".
+                                   format(ver[comp], comp))
                     break
     return paths
 
+
+def get_defined_components(duthost, defined_fw, chassis):
+    """
+    Update the component content if there is definition for the specific host.
```

Comment:
Update the component content, in case there is a pre-definition for a specific host.
Sometimes, if there is some DUTs has specific component(for example a respined board which requires a different CPLD) - it can be defined in the firmware.json file

Reply:
I suggest also to change the firmware.json to include such an instance, as an example

Reply:
Updated